src/test/compile-fail/immut-function-arguments.rs

```diff
@@ -8,14 +8,17 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+// revisions: ast mir
+//[mir]compile-flags: -Z borrowck=mir
 
 fn f(y: Box<isize>) {
-    *y = 5; //~ ERROR cannot assign
+    *y = 5; //[ast]~ ERROR cannot assign
+            //[mir]~^ ERROR cannot assign twice
```

Comment:
Why are we reporting "cannot assign twice" here? That is, I don't see a second assignment? I guess it is referring to whatever initialize `*y` in the caller?

Reply:
So, I think this is a pretty confusing error message, but I see that [this is the current behavior](https://play.rust-lang.org/?gist=f0ba6f96f906250b078b5c67af6fc72c&version=stable). I think we should special-case arguments, but it seems like a separate PR.

Reply:
Filed https://github.com/rust-lang/rust/issues/46659